tests/test_cases/test_cocotb/test_scheduler.py

```diff
@@ -178,3 +178,26 @@ def null_coroutine():
         yield null_coroutine()
 
     yield Timer(100)
+
+
+@cocotb.test()
+async def test_kill_coroutine_waiting_on_the_same_trigger(dut):
+    # gh-1348
+    # NOTE: this test depends on scheduling priority, but the fix for gh-1348
+    # should prevent that from mattering
+    dut.clk.setimmediatevalue(0)
+
+    async def victim():
+        await Timer(1)  # prevent scheduling of RisingEdge before killer
+        await RisingEdge(dut.clk)
+        raise RuntimeError
```

Comment:
Possibly clearer as
```
nonlocal victim_resumed
victim_resumed = True
```

and then doing `assert not victim_resumed` below.
This means we don't have to rely on the exception-in-forked-coroutine semantics to catch the failure.